core/federated/RTI/rti_local.c

```diff
@@ -0,0 +1,206 @@
+/**
+ * @file
+ * @author Erling Jellum (erling.r.jellum@ntnu.no)
+ * @author Edward A. Lee (eal@berkeley.edu)
+ * @author Chadlia Jerad (chadlia.jerad@ensi-uma.tn)
+ * @author Soroush Bateni (soroush@utdallas.edu)
+ * @copyright (c) 2020-2023, The University of California at Berkeley
+ * License in [BSD 2-clause](https://github.com/lf-lang/reactor-c/blob/main/LICENSE.md)
+*/
+
+#ifdef LF_ENCLAVES
+#include "rti_local.h"
+#include "rti_common.h"
+#include "util.h"
+#include "platform.h"
+#include "environment.h"
+#include "trace.h"
+#include "reactor.h"
+
+
+// The following are macros for to be used in the enter and exit
+// of the local RTI API. We extract this to macros to make sure that 
+// mutexes are acquired and released in the correct order.
+#define LTC_LOCKED_PROLOGUE(enclave) \
+    do { \
+        lf_mutex_unlock(&enclave->env->mutex); \
+    } while(0)
+
+#define LTC_LOCKED_EPILOGUE(enclave) \
+    do { \
+        lf_mutex_lock(&enclave->env->mutex); \
+    } while(0)
+
+#define NET_LOCKED_PROLOGUE(enclave) \
+    do { \
+        lf_mutex_unlock(&enclave->env->mutex); \
+        lf_mutex_lock(rti_local->base.mutex); \
+    } while(0)
+
+#define NET_LOCKED_EPILOGUE(enclave) \
+    do { \
+        lf_mutex_unlock(rti_local->base.mutex); \
+        lf_mutex_lock(&enclave->env->mutex); \
+    } while(0)
+#define UPDATE_OTHER_PROLOGUE() \
```

Comment:
Same question regarding `rti_update_other_net_locked`.
These macros don't seem to me to ensure anything about order and just obscure what is going on.

Reply:
I get your point. It is not about ensuring things, but refactoring out these few critical lines so that there is zero chance of someone else, by mistake reordering them while doing some changes to the function. If you find it confusing we can remove the macros and do the locking and unlocking in "plaintext" in the function